inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -494,14 +498,38 @@ bool pass::Serialize::run_on_function(std::shared_ptr<ngraph::Function> f) {
 
     // create xml file
     std::ofstream xml_file(m_xmlPath, std::ios::out);
+    NGRAPH_CHECK(xml_file, "Can't open xml file: \"" + m_xmlPath + "\"");
     xml_doc.save(xml_file);
-
-    // create bin file
-    std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
-    bin_file.write(reinterpret_cast<const char*>(constants.data()),
-                   constants.size() * sizeof(constants[0]));
+    xml_file.flush();
+    bin_file.flush();
 
     // Return false because we didn't change nGraph Function
     return false;
 }
+
+namespace {
+std::string bestBinPath(const std::string &xmlPath, const std::string &binPath) {
```

Comment:
@pelszkow please explain why you need this.

Reply:
Because some e2e tests are failing. Some test pass empty path here.

https://github.com/openvinotoolkit/openvino/blob/e9b89b0bc5b3620b4e2537f3fb56ba6a8f0ff495/inference-engine/include/cpp/ie_cnn_network.h#L243

Reply:
@ilyachur what is expected behaviour of InferenceEngine::CNNNetwork::serialize() when binPath is empty? It's not really [specified](https://docs.openvinotoolkit.org/latest/classInferenceEngine_1_1CNNNetwork.html#aef34988cafc7d34889d2a41e5f4810e5).

Reply:
I propose that we need to create the `bin` file with the same name as `xml` file if `bin` path is empty.

Reply:
Ok. So current approach is OK - .xml is replaced with .bin. 

There is one issue though, we don't check that xml path is valid, and malicious user can pass shorter than 3 character string and we will have out of bounds access. 

I  would do what follows:
_if (xml_path is not valid path (non-empty + alphanumeric characters only?) ) 
   throw
else if (xml_path ends with .xml)
   bin_path =  xml_path with replaced .xml with .bin 
else 
  bin_name = xml_path + ".bin"_

Reply:
I think we should have the next logic:
```
if (bin_path is empty) {
    if (xml_path is not valid path (non-empty + alphanumeric characters only?) )
        throw
    else if (xml_path ends with .xml)
        bin_path = xml_path with replaced .xml with .bin
    else
        // Looks like if xml_path doesn't have .xml as an extension it is not valid path
        throw
}

```

Reply:
I have just realize that we are discuss about `filename` not about whole path (with filename).

Reply:
@pelszkow Did you fix the issue?

Reply:
yes, I drop `isalnum` path validation idea. Without this validation (is not present in master) all works fine.